src/vmm/src/device_manager/mmio.rs

```diff
@@ -110,17 +108,32 @@ impl MMIODeviceManager {
     }
 
     /// Allocates resources for a new device to be added.
-    fn allocate_new_slot(&mut self) -> Result<MMIODeviceInfo> {
-        if self.irq > self.last_irq {
-            return Err(Error::IrqsExhausted);
+    fn allocate_new_slot(&mut self, irq_count: usize) -> Result<MMIODeviceInfo> {
+        let slot: MMIODeviceInfo;
+        match irq_count {
+            0 => {
+                slot = MMIODeviceInfo {
+                    addr: self.mmio_base,
+                    len: MMIO_LEN,
+                    irqs: vec![0],
```

Comment:
Don't trick into thinking it has `irq 0`. Just give it an empty vec:
```suggestion
                    irqs: vec![],
```